src/rgw/rgw_op.cc

```diff
@@ -5917,10 +5899,13 @@ void RGWListMultipart::execute()
   mp.init(s->object.name, upload_id);
   meta_oid = mp.get_meta();
 
-  op_ret = get_multipart_info(store, s, meta_oid, &policy, nullptr, nullptr);
+  multipart_upload_info upload_info;
+  op_ret = get_multipart_info(store, s, meta_oid, &policy, nullptr, &upload_info);
   if (op_ret < 0)
     return;
 
+  s->dest_placement = upload_info.dest_placement;
```

Comment:
@joke-lee the better way to do it would be probably by modifying the list_multipart_parts() interface to include placement info, instead of changing req_state.